### PR TITLE
fix sorting of insert statements

### LINKED ISSUE #561: Keep track of HHH-14344 about insert statement  ordering issue
See this potential fix: https://github.com/hibernate/hibernate-orm/pull/3660#

I haven't looked at the details, but we might want to adapt the test for reactive and review the ActionQueue class.

### PR DESCRIPTION
**Pull request by blafond**
Fixes #561

- fixed sorting of insert statements in reaction action queue
- added utility to capture sql statements via logger
- test compares expected ordered insert sql with actual

### CODE REVIEW

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/InsertOrderingReferenceSeveralDifferentSubclassTest.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+			"insert into UnrelatedEntity (unrelatedValue, id) values ($1, $2)",
+			"insert into BaseEntity (name, TYPE, id) values ($1, 'ZERO', $2)",
+			"insert into BaseEntity (name, PARENT_ID, TYPE, id) values ($1, $2, 'TWO', $3)",
+			"insert into BaseEntity (name, PARENT_ID, TYPE, id) values ($1, $2, 'ONE', $3)"
+	};
+
+	@Override
+	protected Configuration constructConfiguration() {
+		Configuration configuration = super.constructConfiguration();
+		configuration.setProperty( Settings.ORDER_INSERTS, "true" );
```
Comment (DavideD) [inline hibernate-reactive-core/src/test/java/org/hibernate/reactive/InsertOrderingReferenceSeveralDifferentSubclassTest.java:56]: Could you also test what happens when `ORDER_INSERTS` is false?

**Review by DavideD — DISMISSED** [review]
No need for a separate commit with the refactoring of a class that you have just added in this PR.
Just send a PR with the final class, please.

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/testing/SqlStatementTracker.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	public static Predicate<String> DELETE_FILTER = s -> ( s.toLowerCase().startsWith( "delete" ) );
+	public static Predicate<String> SELECT_FILTER = s -> ( s.toLowerCase().startsWith( "select" ) );
+
+	private Predicate<String> filter;
+
+	private List<String> sqlList = new ArrayList<>();
+	private static SqlStatementTracker sqlTracker;
+
+	public SqlStatementTracker(Predicate<String> predicate) {
+		this.sqlTracker = this;
```
[the remark this answers is not in the document]
Comment (blafond) [reply hibernate-reactive-core/src/test/java/org/hibernate/reactive/testing/SqlStatementTracker.java:38]: The `registerService()` method ended up with some inner classes.  `JdbcServicesLogger` implements `getStatementLogger()` so it needed access to the tracker itself.  This works, but I'm not sure if there's a simpler/better pattern or not.

**Comment by DavideD** [comment]
Merged https://github.com/hibernate/hibernate-reactive/pull/1211

With a hcouple of changes to the javadocs

**Pull request closed without merging**